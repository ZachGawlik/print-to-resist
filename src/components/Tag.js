import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Tag.css';

class Tag extends React.Component {
  _onClick = () => {
    this.props.onClick(this.props.name);
  }

  render() {
    if (this.props.onClick) {
      return (
        <span className="Tag Tag--clickable" onClick={this._onClick}>
          {this.props.name}
        </span>
      );
    }
    return <span className="Tag">{this.props.name}</span>;
  }
}

Tag.propTypes = {
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

export default Tag;
