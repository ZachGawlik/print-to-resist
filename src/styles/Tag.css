.Tag {
  margin-right: .5em;
  padding: 0 .5em;
  border-top: 1px dashed transparent;
  border-bottom: 1px dashed transparent;
  border-left: 1px dashed grey;
  border-right: 1px dashed grey;
}

.Tag:focus {
  outline: 0;
  border-left-style: solid;
  border-left-color: black;
  border-right-style: solid;
  border-right-color: black;
}

.Tag--clickable {
  cursor: pointer;
}

.Tag--clickable:hover {
  border-top-color: grey;
  border-bottom-color: grey;
  border-left-color: transparent;
  border-right-color: transparent;
}
