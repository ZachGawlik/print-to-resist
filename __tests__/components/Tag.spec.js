import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Tag from '../../src/components/Tag';

describe('Tag', () => {
  it('renders unclickable tag', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Tag name="test" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders clickable tag', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Tag name="test" onClick={jest.fn()} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
