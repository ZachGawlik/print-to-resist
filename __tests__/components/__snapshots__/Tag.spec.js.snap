// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders clickable tag 1`] = `
<button
  className="Tag Tag--clickable"
  onClick={[Function]}
>
  test
</button>
`;

exports[`Tag renders unclickable tag 1`] = `
<span
  className="Tag"
>
  test
</span>
`;
